app/controllers/incomeSources/manage/ManageObligationsController.scala

```diff
@@ -0,0 +1,238 @@
+ * limitations under the License.
+ */
+
+package controllers.incomeSources.manage
+
+import auth.MtdItUser
+import config.featureswitch.{FeatureSwitching, IncomeSources}
+import config.{AgentItvcErrorHandler, FrontendAppConfig, ItvcErrorHandler}
+import controllers.agent.predicates.ClientConfirmedController
+import controllers.predicates._
+import models.incomeSourceDetails.TaxYear
+import models.incomeSourceDetails.TaxYear.getTaxYearStartYearEndYear
+import play.api.Logger
+import play.api.mvc._
+import services.{DateService, IncomeSourceDetailsService, NextUpdatesService}
+import uk.gov.hmrc.auth.core.AuthorisedFunctions
+import uk.gov.hmrc.http.HeaderCarrier
+import views.html.incomeSources.manage.{ManageIncomeSources, ManageObligations}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+sealed trait Mode {
+  val key: String
+}
+
+case object SelfEmployment extends Mode {
+  override val key = "SE"
+}
+
+case object UkProperty extends Mode {
+  override val key = "UK"
+}
+
+case object ForeignProperty extends Mode {
+  override val key = "FP"
+}
+
+class ManageObligationsController @Inject()(val checkSessionTimeout: SessionTimeoutPredicate,
+                                            val authenticate: AuthenticationPredicate,
+                                            val authorisedFunctions: AuthorisedFunctions,
+                                            val retrieveNino: NinoPredicate,
+                                            val retrieveIncomeSources: IncomeSourceDetailsPredicate,
+                                            val itvcErrorHandler: ItvcErrorHandler,
+                                            implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                            val incomeSourceDetailsService: IncomeSourceDetailsService,
+                                            val retrieveBtaNavBar: NavBarPredicate,
+                                            val obligationsView: ManageObligations,
+                                            nextUpdatesService: NextUpdatesService)
+                                           (implicit val ec: ExecutionContext,
+                                            implicit override val mcc: MessagesControllerComponents,
+                                            val appConfig: FrontendAppConfig) extends ClientConfirmedController
+  with FeatureSwitching {
+
+
+  def showSelfEmployment(changeTo: String, taxYear: String, id: String): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit user =>
+      handleRequest(
+        mode = SelfEmployment,
+        isAgent = false,
+        taxYear,
+        changeTo,
+        id
+      )
+  }
+
+  def showAgentSelfEmployment(changeTo: String, taxYear: String, id: String): Action[AnyContent] = Authenticated.async {
```

Comment:
rename id to incomeSourceId as this is not exposed to UI we free to use required naming convention; this would simplify code support